server/src/test/java/org/opensearch/extensions/ExtensionsOrchestratorTests.java

```diff
@@ -260,12 +269,14 @@ public void testExtensionsDiscovery() throws Exception {
                     "fakeClass2",
                     new ArrayList<String>(),
                     true
-                )
+                ),
+                List.of(new ExtensionDependency("uniqueId0", Version.fromString("2.0.0")))
             )
         );
         assertEquals(expectedExtensionsList.size(), extensionsOrchestrator.extensionIdMap.values().size());
         assertTrue(expectedExtensionsList.containsAll(extensionsOrchestrator.extensionIdMap.values()));
         assertTrue(extensionsOrchestrator.extensionIdMap.values().containsAll(expectedExtensionsList));
+        assertEquals(expectedExtensionsList.get(expectedExtensionsList.size() - 1), expectedDependency);
```

Comment:
These assertions should be (expected, actual), so you want to reverse the order of these.  Also:
 - You know there are exactly two extensions in the list so rather than calculating size-1 just use `get(1)`
 - That getter returns a `DiscoveryExtension` but you need to add the getter for the dependency list to compare vs. the dependency.